
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../generate-callable-loci/">
      
      
        <link rel="next" href="../collect-depth-data/">
      
      
        
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.2">
    
    
      
        <title>Calculate Statistics - clam</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#calculate-statistics" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="clam" class="md-header__button md-logo" aria-label="clam" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            clam
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Calculate Statistics
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9zM20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zm-9.15 3.96h2.3L12 9z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/cademirch/clam" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="clam" class="md-nav__button md-logo" aria-label="clam" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    clam
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/cademirch/clam" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    clam
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            
  
    clam
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../explanation/why-clam/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Why clam?
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../explanation/concepts/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Core Concepts
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Tutorial
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Tutorial
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/01_overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    How-to Guides
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    How-to Guides
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../generate-callable-loci/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Generate Callable Loci
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Calculate Statistics
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Calculate Statistics
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prerequisites
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#input-requirements" class="md-nav__link">
    <span class="md-ellipsis">
      
        Input Requirements
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Input Requirements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#vcf-file" class="md-nav__link">
    <span class="md-ellipsis">
      
        VCF File
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#callable-sites" class="md-nav__link">
    <span class="md-ellipsis">
      
        Callable Sites
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#basic-usage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Basic Usage
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Basic Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fixed-size-windows" class="md-nav__link">
    <span class="md-ellipsis">
      
        Fixed-Size Windows
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#custom-regions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Custom Regions
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#specifying-populations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Specifying Populations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Specifying Populations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#automatic-from-callable-zarr-recommended" class="md-nav__link">
    <span class="md-ellipsis">
      
        Automatic from Callable Zarr (Recommended)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-samples" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using --samples
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-population-file-deprecated" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using --population-file (Deprecated)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#handling-missing-samples" class="md-nav__link">
    <span class="md-ellipsis">
      
        Handling Missing Samples
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-roh-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using ROH Data
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filtering-chromosomes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Filtering Chromosomes
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Filtering Chromosomes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#exclude-specific-chromosomes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Exclude Specific Chromosomes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#include-only-specific-chromosomes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Include Only Specific Chromosomes
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-files" class="md-nav__link">
    <span class="md-ellipsis">
      
        Output Files
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#performance" class="md-nav__link">
    <span class="md-ellipsis">
      
        Performance
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#complete-example" class="md-nav__link">
    <span class="md-ellipsis">
      
        Complete Example
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#working-with-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Working with Results
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Working with Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#python" class="md-nav__link">
    <span class="md-ellipsis">
      
        Python
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r" class="md-nav__link">
    <span class="md-ellipsis">
      
        R
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../collect-depth-data/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Collect Depth Data
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../reference/cli/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CLI Reference
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../reference/input-formats/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Input Formats
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../reference/output-formats/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Output Formats
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prerequisites
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#input-requirements" class="md-nav__link">
    <span class="md-ellipsis">
      
        Input Requirements
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Input Requirements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#vcf-file" class="md-nav__link">
    <span class="md-ellipsis">
      
        VCF File
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#callable-sites" class="md-nav__link">
    <span class="md-ellipsis">
      
        Callable Sites
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#basic-usage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Basic Usage
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Basic Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fixed-size-windows" class="md-nav__link">
    <span class="md-ellipsis">
      
        Fixed-Size Windows
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#custom-regions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Custom Regions
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#specifying-populations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Specifying Populations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Specifying Populations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#automatic-from-callable-zarr-recommended" class="md-nav__link">
    <span class="md-ellipsis">
      
        Automatic from Callable Zarr (Recommended)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-samples" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using --samples
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-population-file-deprecated" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using --population-file (Deprecated)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#handling-missing-samples" class="md-nav__link">
    <span class="md-ellipsis">
      
        Handling Missing Samples
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-roh-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using ROH Data
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filtering-chromosomes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Filtering Chromosomes
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Filtering Chromosomes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#exclude-specific-chromosomes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Exclude Specific Chromosomes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#include-only-specific-chromosomes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Include Only Specific Chromosomes
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-files" class="md-nav__link">
    <span class="md-ellipsis">
      
        Output Files
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#performance" class="md-nav__link">
    <span class="md-ellipsis">
      
        Performance
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#complete-example" class="md-nav__link">
    <span class="md-ellipsis">
      
        Complete Example
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#working-with-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Working with Results
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Working with Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#python" class="md-nav__link">
    <span class="md-ellipsis">
      
        Python
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r" class="md-nav__link">
    <span class="md-ellipsis">
      
        R
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/cademirch/clam/edit/master/docs/how-to/calculate-statistics.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg>
    </a>
  
  


<h1 id="calculate-statistics">Calculate Statistics</h1>
<p>This guide covers how to use <code>clam stat</code> to calculate population genetic statistics from a VCF using callable site information.</p>
<h2 id="prerequisites">Prerequisites</h2>
<ul>
<li>A VCF file (bgzipped and tabix indexed)</li>
<li>Callable loci Zarr from <code>clam loci</code> (recommended)</li>
<li>Optional: population file or <code>--samples</code> TSV for d<sub>xy</sub> and F<sub>ST</sub> (not needed if callable zarr already contains population metadata)</li>
<li>Optional: ROH file for calculating non-ROH heterozygosity</li>
</ul>
<h2 id="input-requirements">Input Requirements</h2>
<h3 id="vcf-file">VCF File</h3>
<p>The input VCF must be bgzipped and tabix indexed:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>bgzip<span class="w"> </span>variants.vcf
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>tabix<span class="w"> </span>-p<span class="w"> </span>vcf<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<h3 id="callable-sites">Callable Sites</h3>
<p>While optional, providing callable sites from <code>clam loci</code> is strongly recommended for accurate statistics:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<p>Without callable sites, all positions in the VCF are assumed callable, which can bias diversity estimates if you have missing data.</p>
<h2 id="basic-usage">Basic Usage</h2>
<h3 id="fixed-size-windows">Fixed-Size Windows</h3>
<p>Calculate statistics in fixed-size windows across the genome:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<p>This calculates Ï€ (and d<sub>xy</sub>, F<sub>ST</sub> if populations are defined) in 10kb windows.</p>
<h3 id="custom-regions">Custom Regions</h3>
<p>Calculate statistics for specific regions defined in a BED file:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>--regions-file<span class="w"> </span>genes.bed<span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<h2 id="specifying-populations">Specifying Populations</h2>
<p>To calculate between-population statistics (d<sub>xy</sub> and F<sub>ST</sub>), clam needs population assignments. There are three ways to provide them:</p>
<h3 id="automatic-from-callable-zarr-recommended">Automatic from Callable Zarr (Recommended)</h3>
<p>If populations were defined during the <code>clam loci</code> step, they are stored in the callable Zarr metadata. <code>clam stat</code> reads them automatically -- no extra flags needed:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<p>This produces <code>dxy.tsv</code> and <code>fst.tsv</code> automatically when the callable zarr contains multiple populations.</p>
<h3 id="using-samples">Using --samples</h3>
<p>Provide a samples TSV to define (or override) population assignments:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>-s<span class="w"> </span>samples.tsv<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<p>Only the <code>sample_name</code> and <code>population</code> columns are used; <code>file_path</code> is ignored for <code>clam stat</code>.</p>
<h3 id="using-population-file-deprecated">Using --population-file (Deprecated)</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>-p<span class="w"> </span>populations.tsv<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<p>The population file maps samples to populations:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>sample1 PopA
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>sample2 PopA
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>sample3 PopB
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>sample4 PopB
</span></code></pre></div>
<div class="admonition warning">
<p class="admonition-title">Population Consistency</p>
<p>When using <code>--samples</code> or <code>--population-file</code> with a callable Zarr that has <code>PopulationCounts</code> type, the number of populations must match the number of population columns in the Zarr. If they don't match, clam will produce a clear error. Use callable sites generated with <code>--per-sample</code> mode for maximum flexibility with different population definitions at stat time.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Sample Names</p>
<p>Sample names must exactly match those in your VCF header.</p>
</div>
<h3 id="handling-missing-samples">Handling Missing Samples</h3>
<p>By default, clam errors if the VCF contains samples not in the population file. To instead warn and exclude those samples:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>-p<span class="w"> </span>populations.tsv<span class="w"> </span>--force-samples<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<h2 id="using-roh-data">Using ROH Data</h2>
<p>If you have runs of homozygosity (ROH) calls, clam can calculate non-ROH heterozygosity by excluding samples within ROH regions at each site:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>-r<span class="w"> </span>roh.bed.gz<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<p>Non-ROH heterozygosity can serve as a proxy for the inbreeding load in a population (<a href="https://www.sciencedirect.com/science/article/pii/S016953472500182X">Kardos et al. 2025</a>).</p>
<p>The ROH file must be:</p>
<ul>
<li>BED format with sample name in the 4th column</li>
<li>bgzipped and tabix indexed</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1"># Prepare ROH file</span>
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>sort<span class="w"> </span>-k1,1<span class="w"> </span>-k2,2n<span class="w"> </span>roh.bed<span class="w"> </span>&gt;<span class="w"> </span>roh.sorted.bed
</span><span id="__span-10-3"><a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>bgzip<span class="w"> </span>roh.sorted.bed
</span><span id="__span-10-4"><a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a>tabix<span class="w"> </span>-p<span class="w"> </span>bed<span class="w"> </span>roh.sorted.bed.gz
</span></code></pre></div>
<p>When ROH data is provided, the <code>heterozygosity.tsv</code> output includes additional columns for ROH-excluded statistics (<code>het_not_in_roh</code>, <code>callable_not_in_roh</code>, <code>heterozygosity_not_in_roh</code>).</p>
<p>See <a href="../../reference/input-formats/#roh-file">Input Formats</a> for file format details.</p>
<h2 id="filtering-chromosomes">Filtering Chromosomes</h2>
<h3 id="exclude-specific-chromosomes">Exclude Specific Chromosomes</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>-x<span class="w"> </span>chrM,chrY<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<h3 id="include-only-specific-chromosomes">Include Only Specific Chromosomes</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>-i<span class="w"> </span>chr1,chr2,chr3<span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<h2 id="output-files">Output Files</h2>
<p><code>clam stat</code> produces tab-separated files in the output directory:</p>
<table>
<thead>
<tr>
<th>File</th>
<th>Description</th>
<th>Generated</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>pi.tsv</code></td>
<td>Nucleotide diversity per population</td>
<td>Always</td>
</tr>
<tr>
<td><code>dxy.tsv</code></td>
<td>Absolute divergence between populations</td>
<td>When &gt;1 population</td>
</tr>
<tr>
<td><code>fst.tsv</code></td>
<td>Fixation index between populations</td>
<td>When &gt;1 population</td>
</tr>
<tr>
<td><code>heterozygosity.tsv</code></td>
<td>Per-sample or per-population heterozygosity</td>
<td>When callable sites provided</td>
</tr>
</tbody>
</table>
<p>The <code>heterozygosity.tsv</code> file contains per-sample heterozygosity when using per-sample callable masks (<code>--per-sample</code> in <code>clam loci</code>), or per-population heterozygosity when using population counts. When ROH data is provided, additional columns report heterozygosity excluding samples in ROH regions.</p>
<p>See <a href="../../reference/output-formats/#clam-stat-output">Output Formats</a> for column descriptions.</p>
<h2 id="performance">Performance</h2>
<p>Use multiple threads for faster processing:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>-t<span class="w"> </span><span class="m">16</span><span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<p>The chunk size affects parallelization and memory usage:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a>clam<span class="w"> </span>stat<span class="w"> </span>-o<span class="w"> </span>results/<span class="w"> </span>-w<span class="w"> </span><span class="m">10000</span><span class="w"> </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span>--chunk-size<span class="w"> </span><span class="m">500000</span><span class="w"> </span>variants.vcf.gz
</span></code></pre></div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If you used <code>clam loci</code> with a specific chunk size, <code>clam stat</code> automatically uses the same chunk size when reading the callable Zarr.</p>
</div>
<h2 id="complete-example">Complete Example</h2>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="c1"># Calculate pi, dxy, and Fst in 50kb windows</span>
</span><span id="__span-15-2"><a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="c1"># - Use callable sites from loci</span>
</span><span id="__span-15-3"><a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a><span class="c1"># - Define two populations</span>
</span><span id="__span-15-4"><a id="__codelineno-15-4" name="__codelineno-15-4" href="#__codelineno-15-4"></a><span class="c1"># - Calculate non-ROH heterozygosity</span>
</span><span id="__span-15-5"><a id="__codelineno-15-5" name="__codelineno-15-5" href="#__codelineno-15-5"></a><span class="c1"># - Exclude mitochondria</span>
</span><span id="__span-15-6"><a id="__codelineno-15-6" name="__codelineno-15-6" href="#__codelineno-15-6"></a><span class="c1"># - Use 16 threads</span>
</span><span id="__span-15-7"><a id="__codelineno-15-7" name="__codelineno-15-7" href="#__codelineno-15-7"></a>clam<span class="w"> </span>stat<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-8"><a id="__codelineno-15-8" name="__codelineno-15-8" href="#__codelineno-15-8"></a><span class="w">  </span>-o<span class="w"> </span>results/<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-9"><a id="__codelineno-15-9" name="__codelineno-15-9" href="#__codelineno-15-9"></a><span class="w">  </span>-w<span class="w"> </span><span class="m">50000</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-10"><a id="__codelineno-15-10" name="__codelineno-15-10" href="#__codelineno-15-10"></a><span class="w">  </span>-c<span class="w"> </span>callable.zarr<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-11"><a id="__codelineno-15-11" name="__codelineno-15-11" href="#__codelineno-15-11"></a><span class="w">  </span>-p<span class="w"> </span>populations.tsv<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-12"><a id="__codelineno-15-12" name="__codelineno-15-12" href="#__codelineno-15-12"></a><span class="w">  </span>-r<span class="w"> </span>roh.bed.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-13"><a id="__codelineno-15-13" name="__codelineno-15-13" href="#__codelineno-15-13"></a><span class="w">  </span>-x<span class="w"> </span>chrM<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-14"><a id="__codelineno-15-14" name="__codelineno-15-14" href="#__codelineno-15-14"></a><span class="w">  </span>-t<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-15"><a id="__codelineno-15-15" name="__codelineno-15-15" href="#__codelineno-15-15"></a><span class="w">  </span>variants.vcf.gz
</span></code></pre></div>
<h2 id="working-with-results">Working with Results</h2>
<h3 id="python">Python</h3>
<div class="language-python highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>
</span><span id="__span-16-3"><a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a><span class="n">pi</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;results/pi.tsv&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
</span><span id="__span-16-4"><a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a><span class="n">dxy</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;results/dxy.tsv&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
</span><span id="__span-16-5"><a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a><span class="n">fst</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;results/fst.tsv&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
</span><span id="__span-16-6"><a id="__codelineno-16-6" name="__codelineno-16-6" href="#__codelineno-16-6"></a>
</span><span id="__span-16-7"><a id="__codelineno-16-7" name="__codelineno-16-7" href="#__codelineno-16-7"></a><span class="c1"># Plot pi across chromosome 1</span>
</span><span id="__span-16-8"><a id="__codelineno-16-8" name="__codelineno-16-8" href="#__codelineno-16-8"></a><span class="n">chr1_pi</span> <span class="o">=</span> <span class="n">pi</span><span class="p">[</span><span class="n">pi</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;chr1&quot;</span><span class="p">]</span>
</span><span id="__span-16-9"><a id="__codelineno-16-9" name="__codelineno-16-9" href="#__codelineno-16-9"></a><span class="n">chr1_pi</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s2">&quot;start&quot;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s2">&quot;pi&quot;</span><span class="p">,</span> <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;line&quot;</span><span class="p">)</span>
</span></code></pre></div>
<h3 id="r">R</h3>
<div class="language-r highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
</span><span id="__span-17-2"><a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a>
</span><span id="__span-17-3"><a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a><span class="kc">pi</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">&quot;results/pi.tsv&quot;</span><span class="p">)</span>
</span><span id="__span-17-4"><a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a><span class="n">dxy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">&quot;results/dxy.tsv&quot;</span><span class="p">)</span>
</span><span id="__span-17-5"><a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a><span class="n">fst</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">&quot;results/fst.tsv&quot;</span><span class="p">)</span>
</span><span id="__span-17-6"><a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a>
</span><span id="__span-17-7"><a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a><span class="c1"># Plot pi by population</span>
</span><span id="__span-17-8"><a id="__codelineno-17-8" name="__codelineno-17-8" href="#__codelineno-17-8"></a><span class="nf">ggplot</span><span class="p">(</span><span class="kc">pi</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="kc">pi</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="n">population</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-17-9"><a id="__codelineno-17-9" name="__codelineno-17-9" href="#__codelineno-17-9"></a><span class="w">  </span><span class="nf">geom_line</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-17-10"><a id="__codelineno-17-10" name="__codelineno-17-10" href="#__codelineno-17-10"></a><span class="w">  </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">chrom</span><span class="p">)</span>
</span></code></pre></div>







  
  






                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../generate-callable-loci/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Generate Callable Loci">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Generate Callable Loci
              </div>
            </div>
          </a>
        
        
          
          <a href="../collect-depth-data/" class="md-footer__link md-footer__link--next" aria-label="Next: Collect Depth Data">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Collect Depth Data
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../..", "features": ["toc.follow", "content.action.edit", "navigation.footer", "navigation.sections"], "search": "../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.79ae519e.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>