
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../why-clam/">
      
      
        <link rel="next" href="../../tutorial/01_overview/">
      
      
        
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.2">
    
    
      
        <title>Core Concepts - clam</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#core-concepts" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="clam" class="md-header__button md-logo" aria-label="clam" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            clam
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Core Concepts
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9zM20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zm-9.15 3.96h2.3L12 9z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/cademirch/clam" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="clam" class="md-nav__button md-logo" aria-label="clam" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    clam
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/cademirch/clam" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    clam
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            
  
    clam
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../why-clam/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Why clam?
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Core Concepts
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Core Concepts
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#callable-loci" class="md-nav__link">
    <span class="md-ellipsis">
      
        Callable Loci
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#depth-thresholds" class="md-nav__link">
    <span class="md-ellipsis">
      
        Depth Thresholds
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Depth Thresholds">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#minimum-depth" class="md-nav__link">
    <span class="md-ellipsis">
      
        Minimum Depth
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#maximum-depth" class="md-nav__link">
    <span class="md-ellipsis">
      
        Maximum Depth
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-chromosome-thresholds" class="md-nav__link">
    <span class="md-ellipsis">
      
        Per-Chromosome Thresholds
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sample-level-vs-site-level-filtering" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sample-Level vs Site-Level Filtering
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sample-Level vs Site-Level Filtering">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sample-level-thresholds" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sample-Level Thresholds
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#site-level-thresholds" class="md-nav__link">
    <span class="md-ellipsis">
      
        Site-Level Thresholds
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#populations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Populations
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-clam-workflow" class="md-nav__link">
    <span class="md-ellipsis">
      
        The clam Workflow
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="The clam Workflow">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-generate-callable-loci-clam-loci" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 1: Generate Callable Loci (clam loci)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-calculate-statistics-clam-stat" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 2: Calculate Statistics (clam stat)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#optional-pre-collect-depth-clam-collect" class="md-nav__link">
    <span class="md-ellipsis">
      
        Optional: Pre-collect Depth (clam collect)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#statistics-calculated" class="md-nav__link">
    <span class="md-ellipsis">
      
        Statistics Calculated
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Statistics Calculated">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nucleotide-diversity" class="md-nav__link">
    <span class="md-ellipsis">
      
        Nucleotide Diversity (π)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#absolute-divergence-dxy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Absolute Divergence (dxy)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fixation-index-fst" class="md-nav__link">
    <span class="md-ellipsis">
      
        Fixation Index (FST)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#runs-of-homozygosity-roh" class="md-nav__link">
    <span class="md-ellipsis">
      
        Runs of Homozygosity (ROH)
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#heterozygosity" class="md-nav__link">
    <span class="md-ellipsis">
      
        Heterozygosity
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Heterozygosity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-sample-mode" class="md-nav__link">
    <span class="md-ellipsis">
      
        Per-Sample Mode
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-population-mode" class="md-nav__link">
    <span class="md-ellipsis">
      
        Per-Population Mode
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#roh-excluded-heterozygosity" class="md-nav__link">
    <span class="md-ellipsis">
      
        ROH-Excluded Heterozygosity
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Tutorial
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Tutorial
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/01_overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    How-to Guides
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    How-to Guides
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../how-to/generate-callable-loci/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Generate Callable Loci
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../how-to/calculate-statistics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Calculate Statistics
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../how-to/collect-depth-data/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Collect Depth Data
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../reference/cli/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CLI Reference
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../reference/input-formats/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Input Formats
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../reference/output-formats/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Output Formats
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#callable-loci" class="md-nav__link">
    <span class="md-ellipsis">
      
        Callable Loci
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#depth-thresholds" class="md-nav__link">
    <span class="md-ellipsis">
      
        Depth Thresholds
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Depth Thresholds">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#minimum-depth" class="md-nav__link">
    <span class="md-ellipsis">
      
        Minimum Depth
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#maximum-depth" class="md-nav__link">
    <span class="md-ellipsis">
      
        Maximum Depth
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-chromosome-thresholds" class="md-nav__link">
    <span class="md-ellipsis">
      
        Per-Chromosome Thresholds
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sample-level-vs-site-level-filtering" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sample-Level vs Site-Level Filtering
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sample-Level vs Site-Level Filtering">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sample-level-thresholds" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sample-Level Thresholds
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#site-level-thresholds" class="md-nav__link">
    <span class="md-ellipsis">
      
        Site-Level Thresholds
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#populations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Populations
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-clam-workflow" class="md-nav__link">
    <span class="md-ellipsis">
      
        The clam Workflow
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="The clam Workflow">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-generate-callable-loci-clam-loci" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 1: Generate Callable Loci (clam loci)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-calculate-statistics-clam-stat" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 2: Calculate Statistics (clam stat)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#optional-pre-collect-depth-clam-collect" class="md-nav__link">
    <span class="md-ellipsis">
      
        Optional: Pre-collect Depth (clam collect)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#statistics-calculated" class="md-nav__link">
    <span class="md-ellipsis">
      
        Statistics Calculated
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Statistics Calculated">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nucleotide-diversity" class="md-nav__link">
    <span class="md-ellipsis">
      
        Nucleotide Diversity (π)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#absolute-divergence-dxy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Absolute Divergence (dxy)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fixation-index-fst" class="md-nav__link">
    <span class="md-ellipsis">
      
        Fixation Index (FST)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#runs-of-homozygosity-roh" class="md-nav__link">
    <span class="md-ellipsis">
      
        Runs of Homozygosity (ROH)
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#heterozygosity" class="md-nav__link">
    <span class="md-ellipsis">
      
        Heterozygosity
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Heterozygosity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-sample-mode" class="md-nav__link">
    <span class="md-ellipsis">
      
        Per-Sample Mode
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-population-mode" class="md-nav__link">
    <span class="md-ellipsis">
      
        Per-Population Mode
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#roh-excluded-heterozygosity" class="md-nav__link">
    <span class="md-ellipsis">
      
        ROH-Excluded Heterozygosity
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/cademirch/clam/edit/master/docs/explanation/concepts.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg>
    </a>
  
  


<h1 id="core-concepts">Core Concepts</h1>
<p>This page explains the key concepts behind clam and how they relate to population genetics analysis.</p>
<h2 id="callable-loci">Callable Loci</h2>
<p>A genomic site is considered <strong>callable</strong> for a sample if we have sufficient confidence that we could accurately determine the genotype at that position. In practice, this means the site has adequate sequencing depth.</p>
<p>Sites that are not callable include:</p>
<ul>
<li>Regions with no coverage (unmapped reads)</li>
<li>Regions with very low coverage (unreliable genotype calls)</li>
<li>Regions with extremely high coverage (often repetitive elements or CNVs)</li>
</ul>
<p>clam tracks callable sites at two levels:</p>
<ol>
<li><strong>Sample-level</strong>: Is this site callable for a specific individual?</li>
<li><strong>Site-level</strong>: Is this site callable for the analysis (e.g., callable in enough samples)?</li>
</ol>
<h2 id="depth-thresholds">Depth Thresholds</h2>
<p>clam uses depth thresholds to determine callability. Both minimum and maximum thresholds are important:</p>
<h3 id="minimum-depth">Minimum Depth</h3>
<p>Sites with very low depth have unreliable genotype calls. A heterozygous site with only 2x coverage has a 50% chance of appearing homozygous simply due to sampling.</p>
<p>Common minimum thresholds range from 5x to 15x depending on your confidence requirements and ploidy.</p>
<h3 id="maximum-depth">Maximum Depth</h3>
<p>Extremely high depth often indicates:</p>
<ul>
<li>Repetitive regions where reads from multiple genomic locations map to the same place</li>
<li>Copy number variants (CNVs) or duplications</li>
<li>Systematic mapping artifacts</li>
</ul>
<p>These regions tend to have unreliable variant calls and are typically excluded. Maximum thresholds are often set to 2-3x the mean genome-wide depth.</p>
<h3 id="per-chromosome-thresholds">Per-Chromosome Thresholds</h3>
<p>Some chromosomes may require different thresholds:</p>
<ul>
<li>Sex chromosomes in samples with XY sex determination have half the expected autosomal depth</li>
<li>Organellar genomes (mitochondria, chloroplasts) often have much higher depth</li>
</ul>
<p>clam supports per-chromosome threshold files to handle these cases.</p>
<h2 id="sample-level-vs-site-level-filtering">Sample-Level vs Site-Level Filtering</h2>
<p>clam applies thresholds in two stages:</p>
<h3 id="sample-level-thresholds">Sample-Level Thresholds</h3>
<p>First, for each sample at each position, clam checks if the depth falls within the acceptable range:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>min_depth &lt;= sample_depth &lt;= max_depth
</span></code></pre></div>
<p>If yes, that site is callable for that sample.</p>
<h3 id="site-level-thresholds">Site-Level Thresholds</h3>
<p>Next, clam can apply aggregate filters across all samples:</p>
<ul>
<li><strong>Proportion callable</strong> (<code>-d</code>): What fraction of samples must be callable? Setting <code>-d 0.8</code> requires 80% of samples to pass individual thresholds.</li>
<li><strong>Mean depth range</strong> (<code>--min-mean-depth</code>, <code>--max-mean-depth</code>): What is the acceptable range for mean depth across all samples at a site?</li>
</ul>
<p>These filters help identify sites that are systematically problematic across the dataset.</p>
<h2 id="populations">Populations</h2>
<p>Many population genetic statistics compare diversity within and between groups:</p>
<ul>
<li><strong>π (pi)</strong>: Nucleotide diversity <em>within</em> a population</li>
<li><strong>d<sub>xy</sub></strong>: Absolute divergence <em>between</em> two populations  </li>
<li><strong>F<sub>ST</sub></strong>: Relative differentiation between populations</li>
</ul>
<p>To calculate these statistics, clam needs to know which samples belong to which population. You can specify populations using <code>--samples</code> (recommended) or <code>--population-file</code> (deprecated), or let clam read them automatically from zarr metadata.</p>
<p>When populations are defined:</p>
<ul>
<li><code>clam collect</code> stores population metadata in the output zarr</li>
<li><code>clam loci</code> tracks callable sites per population (how many samples in each population are callable at each site) and stores population metadata in the output zarr</li>
<li><code>clam stat</code> reads population metadata from the callable zarr automatically, or you can override with <code>--samples</code> or <code>-p</code></li>
</ul>
<p>This means populations only need to be specified once -- typically during <code>clam collect</code> or <code>clam loci</code> -- and flow through the pipeline via zarr metadata.</p>
<p>Without population assignments at any stage, clam treats all samples as a single population and only calculates π.</p>
<h2 id="the-clam-workflow">The clam Workflow</h2>
<p>A typical clam analysis has two main steps:</p>
<h3 id="step-1-generate-callable-loci-clam-loci">Step 1: Generate Callable Loci (<code>clam loci</code>)</h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>Depth files (D4/GVCF) → clam loci → Callable loci (Zarr)
</span></code></pre></div>
<p>This step processes depth information and applies your thresholds to determine which sites are callable. The output is a compact Zarr array storing callable counts per population at each genomic position, along with population metadata that downstream commands can read automatically.</p>
<h3 id="step-2-calculate-statistics-clam-stat">Step 2: Calculate Statistics (<code>clam stat</code>)</h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>Callable loci (Zarr) + Variants (VCF) → clam stat → Statistics (TSV)
</span></code></pre></div>
<p>This step combines callable site information with variant calls to compute accurate diversity statistics. The callable loci provide the denominator (total comparisons), while the VCF provides the numerator (differences).</p>
<h3 id="optional-pre-collect-depth-clam-collect">Optional: Pre-collect Depth (<code>clam collect</code>)</h3>
<p>For workflows where you want to run <code>loci</code> multiple times with different thresholds:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>Depth files (D4/GVCF) → clam collect → Depth (Zarr) → clam loci → Callable loci (Zarr)
</span></code></pre></div>
<p>The <code>collect</code> step stores raw depth values in an efficient Zarr format. This is faster than re-reading the original depth files when testing multiple threshold configurations.</p>
<p>Population assignments are preserved through this pipeline -- if you specify populations during <code>collect</code> or <code>loci</code>, they are stored in the Zarr metadata and automatically used by downstream commands.</p>
<h2 id="statistics-calculated">Statistics Calculated</h2>
<p>clam calculates the following statistics in windows across the genome:</p>
<h3 id="nucleotide-diversity">Nucleotide Diversity (π)</h3>
<p>The average number of pairwise differences per site within a population:</p>
<div class="arithmatex">\[\pi = \frac{\sum \text{pairwise differences}}{\sum \text{pairwise comparisons}}\]</div>
<h3 id="absolute-divergence-dxy">Absolute Divergence (d<sub>xy</sub>)</h3>
<p>The average number of pairwise differences per site between two populations:</p>
<div class="arithmatex">\[d_{xy} = \frac{\sum \text{between-population differences}}{\sum \text{between-population comparisons}}\]</div>
<h3 id="fixation-index-fst">Fixation Index (F<sub>ST</sub>)</h3>
<p>A measure of population differentiation, calculated using the Hudson estimator:</p>
<div class="arithmatex">\[F_{ST} = \frac{d_{xy} - \frac{\pi_1 + \pi_2}{2}}{d_{xy}}\]</div>
<p>clam uses a ratio-of-averages approach, summing numerators and denominators across sites within each window before computing the final ratio.</p>
<h2 id="runs-of-homozygosity-roh">Runs of Homozygosity (ROH)</h2>
<p>In populations with recent inbreeding or small effective population size, individuals may have long runs of homozygosity (ROH). When calculating diversity statistics, it can be useful to exclude samples that are within ROH regions at each site.</p>
<p>Non-ROH heterozygosity can serve as a proxy for the inbreeding load in a population. This is because deleterious mutations that were previously masked as heterozygotes become exposed in ROH regions, and the abundance of such mutations scales with genetic diversity (<a href="https://www.sciencedirect.com/science/article/pii/S016953472500182X">Kyriazis et al. 2025</a>).</p>
<p>clam can optionally accept ROH intervals (<code>--roh</code>) and will calculate heterozygosity excluding samples in ROH regions. At each site, any sample falling within an ROH region is excluded from the heterozygosity calculation for that site.</p>
<h2 id="heterozygosity">Heterozygosity</h2>
<p>Heterozygosity is the proportion of heterozygous sites among callable sites:</p>
<div class="arithmatex">\[H = \frac{n_{\text{het}}}{n_{\text{callable}}}\]</div>
<p>Where:</p>
<ul>
<li><span class="arithmatex">\(n_{\text{het}}\)</span> is the number of heterozygous genotypes</li>
<li><span class="arithmatex">\(n_{\text{callable}}\)</span> is the number of callable sites</li>
</ul>
<p>When callable sites are provided, clam outputs a <code>heterozygosity.tsv</code> file with heterozygosity estimates per window.</p>
<h3 id="per-sample-mode">Per-Sample Mode</h3>
<p>When using per-sample callable masks (<code>--per-sample</code> in <code>clam loci</code>), heterozygosity is calculated for each sample individually. This provides the most accurate estimates because each sample has its own callable site mask.</p>
<p>For each sample in each window:</p>
<ul>
<li><code>het_total</code>: Number of heterozygous genotypes for the sample</li>
<li><code>callable_total</code>: Number of sites where the sample is callable</li>
<li><code>heterozygosity</code>: <code>het_total / callable_total</code></li>
</ul>
<h3 id="per-population-mode">Per-Population Mode</h3>
<p>When using population-level callable counts (default <code>clam loci</code> output), heterozygosity is calculated per population by summing across all samples in the population.</p>
<p>For each population in each window:</p>
<ul>
<li><code>het_total</code>: Sum of heterozygous genotypes across all samples in the population</li>
<li><code>callable_total</code>: Sum of callable sites across all samples in the population</li>
<li><code>heterozygosity</code>: <code>het_total / callable_total</code></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Per-population heterozygosity with ROH exclusion is approximate because the callable counts are aggregated at the population level. For accurate per-sample ROH exclusion, use per-sample callable masks.</p>
</div>
<h3 id="roh-excluded-heterozygosity">ROH-Excluded Heterozygosity</h3>
<p>When ROH data is provided, clam also calculates heterozygosity after excluding samples in ROH regions:</p>
<ul>
<li><code>het_not_in_roh</code>: Heterozygous sites where the sample is not in an ROH region</li>
<li><code>callable_not_in_roh</code>: Callable sites where the sample is not in an ROH region</li>
<li><code>heterozygosity_not_in_roh</code>: <code>het_not_in_roh / callable_not_in_roh</code></li>
</ul>







  
  






                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../why-clam/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Why clam?">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Why clam?
              </div>
            </div>
          </a>
        
        
          
          <a href="../../tutorial/01_overview/" class="md-footer__link md-footer__link--next" aria-label="Next: Getting Started">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Getting Started
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../..", "features": ["toc.follow", "content.action.edit", "navigation.footer", "navigation.sections"], "search": "../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.79ae519e.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>